controllers/cluster_controller.go

```diff
@@ -236,14 +237,33 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 		}
 	}
 
-	if err = r.ensureClusterOwnerReferences(ctx, cluster); err != nil {
+	childObjs, err := r.getChildObjectsForCluster(ctx, cluster)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err = r.ensureClusterOwnerReferences(ctx, cluster, childObjs); err != nil {
 		return ctrl.Result{}, err
 	}
 
-	return r.reconcile(ctx, log, cluster)
+	var aggregatedGeneration int64
+	// Compute the aggregated generation by summing up the .metadata.generation value
+	// for all the child objects of this cluster.
+	for _, obj := range childObjs {
+		aggregatedGeneration += obj.GetGeneration()
+	}
+
+	// If there is no difference between the aggregated generation and childrenReconciledGeneration,
+	// and there is no difference in the reconciled generation and .metadata.generation of the cluster,
+	// then return without any further processing.
+	if aggregatedGeneration == cluster.Status.ChildrenReconciledGeneration && cluster.Status.ReconciledGeneration == cluster.Generation {
+		return ctrl.Result{}, nil
```

Comment:
maybe worth adding a log here
```suggestion
        log.Info("Generation and child generation match reconciled generation, skipping reconciliation.")
		return ctrl.Result{}, nil
```